### PR TITLE
[Merged by Bors] - feat: use recommended TSConfig and remove `typeRoots`

### PR DESCRIPTION
**Pull request by EstebanBorai**
Also uses recommended configurations from https://github.com/tsconfig/bases.
Removed [`typeRoots`](https://www.typescriptlang.org/tsconfig/#typeRoots) because we dont include `node` directory, TS must resolve types automatically when they are in the `@types` directory inside the `node_modules`.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -23,17 +11,22 @@
...
     "strict": true,
     "moduleResolution": "node",
-    "typeRoots": [
-      "src",
-      "node_modules/@types",
-      "types"
-    ],
-    "types": [
-      "node"
-    ],
```
Comment (EstebanBorai) [inline tsconfig.json:33]: Allow TS to resolve the types automatically when included in `node_modules/@types`.
We just need to make sure our package types are in the final tarball before releasing.

**Review by digikata — COMMENTED** [review]
Since we have a known issue w/ node 20 should we leave some of these matrix versions at 18 for now?

**Comment by EstebanBorai** [comment]
> Since we have a known issue w/ node 20 should we leave some of these matrix versions at 18 for now?

Yeah! As @morenol comments, for versions newer than `18.14` we will encounter:

```
pointer dereference: address must be a multiple of 0x8 but is 0x3ebb01d4a192'
```

We have to check on bindings to fix this.

**Review by digikata — APPROVED** [review]
LGTM. - I verified these changes  in a hacky way with the distributed version by `npm install @fluvio/client --no-scripts`, hacking tsconfig.json, then manually running `tsc -p .` in the @fluvio/client npm_modules directory.  I could then compile the example.ts

**Comment by digikata** [comment]
bors ping

**Review by digikata — APPROVED** [review]
re-approving to wake up bors?

**Pull request closed without merging**